src/Azure.Functions.Cli/Helpers/AzureHelper.cs

```diff
@@ -211,5 +235,39 @@ private static async Task ArmHttpAsync(HttpMethod method, Uri uri, string access
                     : new HttpResult<Dictionary<string, string>, string>(null, errorMessage);
             }
         }
+
+        public static async Task PrintFunctionsInfo(Site functionApp, string accessToken)
+        {
+            var functions = await GetFunctions(functionApp, accessToken);
+            {
+
+                ColoredConsole.WriteLine(TitleColor($"Functions in {functionApp.SiteName}:"));
+                foreach (var function in functions)
+                {
+                    var trigger = function
+                        .Config?["bindings"]
+                        ?.FirstOrDefault(o => o["type"]?.ToString().IndexOf("Trigger", StringComparison.OrdinalIgnoreCase) != -1)
+                        ?["type"];
+
+                    trigger = trigger ?? "No Trigger Found";
+
+                    ColoredConsole.WriteLine($"    {function.Name} - [{VerboseColor(trigger.ToString())}]");
+                    if (!string.IsNullOrEmpty(function.InvokeUrlTemplate))
+                    {
+                        // If there's a key available, add it to the url
+                        var key = await GetFunctionKey(function.Href.AbsoluteUri, functionApp.ScmUri, accessToken);
+                        if (!string.IsNullOrEmpty(key))
```

Comment:
lets not display the key be default. Add a `--show-keys` flag and only show keys if that's specified